File: python/socketd/socketd/core/SimpleListener.py
from socketd.core.Listener import Listener


class SimpleListener(Listener):

    def on_open(self, session):
        pass

    async def on_message(self, session, message):
        pass

    def on_close(self, session):
        pass

    def on_error(self, session, error):
        pass

File: go/socketd/transport/core/listener/simple_listener.go
package listener

import (
	"socketd/transport/core"
	"socketd/transport/core/message"
)

//var _ core.Listener = new(SimpleListener)

type SimpleListener struct {
}

func (s SimpleListener) OnOpen(session core.Session) error {
	return nil
}

func (s SimpleListener) OnMessage(session core.Session, message *message.Frame) error {
	return nil
}

func (s SimpleListener) OnClose(session core.Session) {

}

func (s SimpleListener) OnError(session core.Session, err error) {

}
